extensions/git/src/model.ts

```diff
@@ -737,6 +793,58 @@ export class Model implements IRemoteSourcePublisherRegistry, IPostCommitCommand
 		return [...this.pushErrorHandlers];
 	}
 
+	private async isRepositoryOutsideWorkspace(repositoryPath: string): Promise<boolean> {
+		if (!workspace.workspaceFolders || workspace.workspaceFolders.length === 0) {
+			return true;
+		}
+
+		const result = await Promise.all(workspace.workspaceFolders.map(async folder => {
+			const workspaceFolderRealPath = await this.getWorkspaceFolderRealPath(folder);
+			return pathEquals(workspaceFolderRealPath, repositoryPath) || isDescendant(workspaceFolderRealPath, repositoryPath);
+		}));
+
+		return !result.some(r => r);
+	}
+
+	private async getWorkspaceFolderRealPath(workspaceFolder: WorkspaceFolder): Promise<string> {
+		if (!this._workspaceFolders.has(workspaceFolder.uri.fsPath)) {
+			const realPath = await fs.promises.realpath(workspaceFolder.uri.fsPath, { encoding: 'utf8' });
+			this._workspaceFolders.set(workspaceFolder.uri.fsPath, realPath);
+		}
+
+		return this._workspaceFolders.get(workspaceFolder.uri.fsPath)!;
```

Comment:
Usually the nicer pattern, since there's no `!` needed and you only access the map once, if the element is already there:

```suggestion
		let result = this._workspaceFolders.get(workspaceFolder.uri.fsPath);
		
		if (!result) {
			result = await fs.promises.realpath(workspaceFolder.uri.fsPath, { encoding: 'utf8' });
			this._workspaceFolders.set(workspaceFolder.uri.fsPath, result);
		}

		return result;
```